pages/doc/integrations_tas_custom_app_metrics_howto.md

```diff
@@ -0,0 +1,75 @@
+
+In the scraping model, the rate of metrics collection can be centrally controlled. In a push model, applications may or may not be able to throttle their metric volume if they are overwhelming the metrics service.
+
+The output of a `/metrics` endpoint in a scraping model is normally human readable, which can help with debugging the metrics coming from a running application.
+
+Pushing metrics can produce more detail; in the push model, a metric can be submitted as often as practically limits allow - possibly multiple times per second. In a scrape model, only changes since the last scrape are seen. Typically metrics data does not need to be extremely high-fidelity in order to be useful, and this periodic sampling is acceptable to most users.
+
+Very short-lived processes **must** use a push model, because they do not run long enough to be discovered and scraped by a metrics collector.
+
+## Metrics Registrar
+
+TAS supports a pull model for custom app metrics using the “Metrics Registrar” component.
+
+The metrics registrar is a TAS component that periodically scrapes registered endpoints and forwards scraped metrics to the Loggregator Firehose. In the Wavefront Nozzle, these metrics are then ingested in the same way as platform or service tile metrics, and become visible in Wavefront automatically.
+
+To instrument an application for metrics registrar scraping, you will typically use either a prometheus SDK or an metrics framework built into your application stack, such as Spring Boot Actuator.
+
+After deploying your application, you register it using the `cf` CLI, as described in the documentation: [https://docs.pivotal.io/application-service/metric-registrar/using.html](https://docs.pivotal.io/application-service/metric-registrar/using.html).
+
+## Wavefront Service Broker
+
+To implement a push-based metrics pattern using Wavefront, you can use the Wavefront SDK for your language stack, or you can use a pluggable abstraction like Spring Boot Actuator.
+
+In either case, you will need to provide your application with a URL for a wavefront proxy instance.
+
+The supported way to do this using the Wavefront Nozzle for TAS is to use the Wavefront Service Broker to create a service instance and bind it to your application. You can then push custom metrics to the proxy from your own application code.
+
+## Factors to Consider
+
+### Spring Boot
+
+Spring Boot is commonly used to build apps running on TAS.
+
+Spring Boot Actuator supports both push and pull models, with the only difference typically being configuration changes. In that case, it becomes relatively easy to switch from push to pull or vice-versa.
+
+### Wavefront Proxy Load
+
+All additional custom app metrics increase the load on the Wavefront Proxy. It may be necessary to increase the count of wavefront proxy instances and to deploy an IaaS load balancer to balance traffic across proxy instances.
```

Comment:
All additional custom app metrics increase the load on the Wavefront proxy. You might have to increase the count of Wavefront proxies and to deploy an IaaS load balancer to balance traffic across proxy instances.